src/Microsoft.AspNet.Mvc.Localization/LocalizedHtmlString.cs

```diff
@@ -26,13 +27,28 @@ public LocalizedHtmlString(string key, string value)
         /// <param name="key">The name of the string resource.</param>
         /// <param name="value">The string resource.</param>
         /// <param name="isResourceNotFound">A flag that indicates if the resource is not found.</param>
-        public LocalizedHtmlString(string key, string value, bool isResourceNotFound)
-            : base(value)
+        /// <param name="arguments">The values to format the <paramref name="value"/> with.</param>
+        public LocalizedHtmlString(string key, string value, bool isResourceNotFound, params object[] arguments)
         {
+            if (key == null)
+            {
+                throw new ArgumentException(nameof(key));
+            }
+
+            if (value == null)
+            {
+                throw new ArgumentException(nameof(value));
+            }
+
             Key = key;
+            Value = value;
             IsResourceNotFound = isResourceNotFound;
+            _arguments = arguments ?? EmptyArguments;
         }
 
+        // Reviewers: The following properties, especially Key and IsResourceNotFound seem to exist only for
+        // consistency with LocalizedString but they're unused. Should they be removed?
```

Comment:
Or, the `Key` was used to throw a meaningful error message when `IsResourceNotFound` was true.